GoogleForCreators/web-stories-wp
# Dashboard: Grid design updates

## Context

Updates my stories and explore templates grid and title areas to [new designs](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=467%3A13745).

## Summary

Namely, new designs are tighter and there's a min width for the dashboard, so updating how the grid resizes while keeping aspect ratio. Otherwise, just swapping out typography components and theme for design system. Cleaning up a bit of dead code as I go, will do a big sweep up after changes are all merged. 

Also fixed story preview dialogs while here, realized those were broken - those are behind a feature flag that could probably be turned on. Will bring this up to Omar and Matt.

| Screenshot | Source | 
| --- | --- | 
| <img width="622" alt="Screen Shot 2021-03-02 at 12 20 13 PM" src="https://user-images.githubusercontent.com/10720454/109702667-c275ce80-7b51-11eb-80da-6b2289ef8878.png"> | Figma specs |
![Screen Shot 2021-03-02 at 12 21 33 PM](https://user-images.githubusercontent.com/10720454/109702734-da4d5280-7b51-11eb-94f9-e09f8a0f1248.png) | main |
![Screen Shot 2021-03-02 at 12 11 39 PM](https://user-images.githubusercontent.com/10720454/109702747-e1746080-7b51-11eb-9d71-c61c2d8d3813.png) | This branch |


## Relevant Technical Choices

- Moved page size theme values to be constants instead to continue to gut the dashboard theme. those don't need to be in the theme. Especially since we've got a forced min width now.
- Added specifications to design system for links as buttons to control hover and focus color - common css (wordPress) hijacks the color otherwise. 😠  
- The grid titles on the stories grid are a headline style _but also_ links so updating the headline style to accept anchors and give them link accessibility that matches the link components. 

## To-do

- Still pending review/QA for context menus and list view for dashboard. Once those are merged we can gut a bunch of stuff!
- Will follow up with template details view.
- Used my best judgement on the hover view of grid items since that's not in figma, suspect there might be some tweaks there. 

## User-facing changes

**Updated Keyboard Nav functionality**
| ![grid keyboard](https://user-images.githubusercontent.com/10720454/109703788-1d5bf580-7b53-11eb-81cb-bcea72d5f8f6.gif)

**Update Explore Templates**
![Screen Shot 2021-03-02 at 12 36 30 PM](https://user-images.githubusercontent.com/10720454/109704555-f9e57a80-7b53-11eb-955f-393aeafef7d8.png)

**RTL 🎉 **
![Screen Shot 2021-03-02 at 12 57 23 PM](https://user-images.githubusercontent.com/10720454/109707014-e5ef4800-7b56-11eb-8ddf-1b9a0d2e0ba7.png)



## Testing Instructions

- My Stories (Regression testing) 
  - On hover see 'Open in Editor' 
  - If show story preview experiment is enabled you should see another button in the middle on hover that says 'Preview'. 
  - If show story preview experiment is enabled and you click on the 'Preview' button a dialog will open that previews the story. 
  - These same buttons should show up if you focus on a grid item with the keyboard (tab until the entire grid has that focus outline, then use your arrow keys)
  - Story titles should remain clickable, they are links that send you to the editor. 
  - Resizing the grid will recalculate space for story previews and keep the grid tight and justified. 
 
- Explore Templates (Regression testing) 
  - Keyboard focus works the same as my stories does as outlined above. 
  - Resizing the grid will recalculate space for template previews and keep the grid tight and justified. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #4964
